stdlib/compare.ml

```diff
@@ -0,0 +1,37 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*   Copyright 2025 Kate Deplaix                                          *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+module Phys = struct
+  external equal : 'a -> 'a -> bool = "%eq"
+
+  module Syntax = struct
+    external ( == ) : 'a -> 'a -> bool = "%eq"
+    external ( != ) : 'a -> 'a -> bool = "%noteq"
+  end
+end
+
+module Poly = struct
+  external equal : 'a -> 'a -> bool = "%equal"
+  external compare : 'a -> 'a -> int = "%compare"
+
+  let min = Stdlib.min
+  let max = Stdlib.max
+
+  module Syntax = struct
+    external ( = ) : 'a -> 'a -> bool = "%equal"
+    external ( <> ) : 'a -> 'a -> bool = "%notequal"
+    external ( < ) : 'a -> 'a -> bool = "%lessthan"
+    external ( > ) : 'a -> 'a -> bool = "%greaterthan"
+    external ( <= ) : 'a -> 'a -> bool = "%lessequal"
+    external ( >= ) : 'a -> 'a -> bool = "%greaterequal"
+  end
```

Comment:
I am in two minds about these submodules.

The main use of them is with numeric types (`Int`, `Int64`, `Char`, etc). If anything, monomorphic functions of these operators should be added to those modules. I don't think the operators  `<`, `>`, `<=`, `>=` are very useful in contexts where one is explicitly using a polymorphic compare (since the ordering is implementation-defined). As for `=` and `<>`, they are useful, but once you have put them inside a submodule, I don't see myself using a local open to save a few character with respect to `Compare.Poly.equal`.

In other words, perhaps it would be simpler not to include the `Syntax` submodules at all (which would also mean that we do not have to discuss what to name the submodule).

Reply:
> In other words, perhaps it would be simpler not to include the Syntax submodules at all (which would also mean that we do not have to discuss what to name the submodule).

Yes, that would also be my preference to reduce decision fatigue.

Reply:
fair enough, that works for me. Done.